azure-kusto-ingest/azure/kusto/ingest/exceptions.py

```diff
@@ -44,3 +44,15 @@ class KustoMissingMappingReferenceError(KustoClientError):
     def __init__(self):
         message = "When stream format is json, mapping name must be provided."
         super(KustoMissingMappingReferenceError, self).__init__(message)
+
+
+class KustoInvalidEndpointError(KustoClientError):
+    """Raised when trying to ingest to invalid cluster type."""
+
+    def __init__(self, expected_service_type, actual_service_type, suggested_endpoint_url=None):
+        message = "You are using '{0}' client type, but the provided endpoint is of ServiceType '{1}'. Initialize the client with the appropriate endpoint URI".format(
+            expected_service_type, actual_service_type
+        )
+        if suggested_endpoint_url is not None:
```

Comment:
if not suggested_endpoint_url:

Reply:
Much of what I did here was based on searching best practices, since this my first Python work. I wrote this particular line according to: https://stackoverflow.com/questions/3965104/not-none-test-in-python
It says that for readability, "is not None" is better.
Should I change it anyway?

Reply:
Unless you say otherwise, I'm leaving it as it seems the Python community prefers it.

Reply:
The reason to use "x is not None" over simple "not x" is some types are falsy, and sometimes you can get the wrong result if you don't compare explicitly. 
Here I think you'd want to use str falsy nature so if the string is empty there is no point to edit the message.

Reply:
You make a good point, thanks for explaining!